utility/module.go

```diff
@@ -44,12 +63,94 @@ func (*utilityModule) Create(bus modules.Bus, options ...modules.ModuleOption) (
 
 	m.config = utilityCfg
 	m.mempool = types.NewTxFIFOMempool(utilityCfg.MaxMempoolTransactionBytes, utilityCfg.MaxMempoolTransactions)
+	m.logger = logger.Global.CreateLoggerForModule(m.GetModuleName())
+
+	return m, enableActorModules(cfg, m, bus)
+}
+
+// enableActorModules enables the actor-specific modules and adds them to the utility module's actorModules to be started later.
+func enableActorModules(cfg *configs.Config, m *utilityModule, bus modules.Bus) error {
+	fishermanCfg := cfg.Fisherman
+	servicerCfg := cfg.Servicer
+	validatorCfg := cfg.Validator
+	portalCfg := cfg.Portal
+
+	if servicerCfg.Enabled {
+		servicer, err := servicer_module.CreateServicer(bus)
+		if err != nil {
+			m.logger.Error().Err(err).Msg("failed to create servicer module")
+			return err
+		}
+		m.servicer = &servicer
+		m.actorModules = append(m.actorModules, servicer)
+	}
+
+	if fishermanCfg.Enabled {
+		fisherman, err := fisherman.CreateFisherman(bus)
+		if err != nil {
+			m.logger.Error().Err(err).Msg("failed to create fisherman module")
+			return err
+		}
+		m.fisherman = &fisherman
+		m.actorModules = append(m.actorModules, fisherman)
+	}
+
+	if validatorCfg.Enabled {
+		validator, err := validator_module.CreateValidator(bus)
+		if err != nil {
+			m.logger.Error().Err(err).Msg("failed to create validator module")
+			return err
+		}
+		m.validator = &validator
+		m.actorModules = append(m.actorModules, validator)
+	}
+
+	if portalCfg.Enabled {
+		portal, err := portal_module.CreatePortal(bus)
+		if err != nil {
+			m.logger.Error().Err(err).Msg("failed to create portal module")
+			return err
+		}
+		m.portal = &portal
+		m.actorModules = append(m.actorModules, portal)
+	}
 
-	return m, nil
+	actors := m.GetActorModuleNames()
+	if len(m.actorModules) > 1 {
```

Comment:
Can you move this into a helper function call `validateActorModuleExclusivity` with a `TODO` comment to figure out all the rules we want to enable?